feat: Add CI/CD Integration Planner web tool

dyb5784 opened the pull request:
- Interactive pipeline visualization with Mermaid.js diagrams
- Goal architect with security objectives and compliance mapping
- Step-by-step action planner with implementation roadmap
- Support for major CI/CD tools (GitHub Actions, Jenkins, GitLab CI, etc.)
- Compliance framework support including CMMC 1.0 and CMMC 2.0
- Responsive design with modern UI/UX
- Pure HTML/CSS/JavaScript implementation with no external dependencies

This tool helps organizations plan and implement in-toto integration
into their CI/CD pipelines with visual guidance and structured planning.

dependabot[bot] committed:
Bump ruff from 0.11.6 to 0.11.8 in the test-and-lint-dependencies group

Bumps the test-and-lint-dependencies group with 1 update: [ruff](https://github.com/astral-sh/ruff).

Updates `ruff` from 0.11.6 to 0.11.8
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.11.6...0.11.8)

---
updated-dependencies:
- dependency-name: ruff
  dependency-version: 0.11.8
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: test-and-lint-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Daniyel Yaacov <daniel.bilar@gmail.com>

dependabot[bot] committed:
Bump ruff from 0.11.8 to 0.11.9 in the test-and-lint-dependencies group

Bumps the test-and-lint-dependencies group with 1 update: [ruff](https://github.com/astral-sh/ruff).

Updates `ruff` from 0.11.8 to 0.11.9
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.11.8...0.11.9)

---
updated-dependencies:
- dependency-name: ruff
  dependency-version: 0.11.9
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: test-and-lint-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Daniyel Yaacov <daniel.bilar@gmail.com>

Commit:
feat: Add CI/CD Integration Planner web tool

Signed-off-by: Daniyel Yaacov <daniel.bilar@gmail.com>

Commit:
feat: README: Added DuckDuckGo web tested and  ISO 27001 Features

Signed-off-by: Daniyel Yaacov <daniel.bilar@gmail.com>

lukpueh commented on in_toto/tools/cicd-planner/index.html:
```diff
@@ -0,0 +1,981 @@
+            width: 100%;
+            height: 100%;
+            background: linear-gradient(90deg, transparent, rgba(255,255,255,0.2), transparent);
+            transition: left 0.5s;Great. Let go dddper. Kindly Flesh out ISO 27001 Compliance framework support
```
Is this an AI prompt spilled into your code??

dyb5784 replied:
IYH Yes. I loaded the base into Claude AI Code and tried to extend it. Good catch. I have a new version of the html code.

dyb5784 replied:
IYH I pushed an update cleaned and expanded

Commit:
# feat: Add in-toto Integration Planner for Securing CI/CD Pipelines

This pull request introduces an in-toto Integration Planner to help teams securely integrate in-toto into their CI/CD pipelines. The planner provides:

- Interactive digital twin pipeline visualization
- Goal architect for defining security objectives
- Step-by-step action plans for implementation
- Domain-specific configurations (general, financial, military, legacy systems)
- Key management strategy setup
- Compliance framework integration

The planner supports multiple CI/CD tools including Jenkins, GitLab CI, GitHub Actions, Azure DevOps, and CircleCI.

Key changes:
- Added HTML-based in-toto CI/CD Integration Planner
- Included Mermaid.js for pipeline visualization
- Added configuration options for various domains and compliance frameworks
- Integrated support for legacy systems

How to review:
1. Test the planner with your CI/CD pipeline configuration
2. Verify the digital twin visualization accurately reflects your pipeline
3. Review the generated action plans for completeness
4. Validate domain-specific configurations match your requirements

Signed-off-by: Daniyel Yaacov <daniel.bilar@gmail.com>

Commit:
The README includes:
Key Sections:

Overview - Clear explanation of what the tool does
Features - Detailed breakdown of the three main components (Digital Twin, Goal Architect, Action Planner)
Supported Integrations - All CI/CD platforms, key management systems, and compliance frameworks
Getting Started - Step-by-step instructions for users
Implementation Workflow - Phased approach with timelines
Technical Architecture - Technology stack and security features
Security Considerations - Best practices for key management and attestation
Contributing Guidelines - How others can contribute to the project

Signed-off-by: Daniyel Yaacov <daniel.bilar@gmail.com>

dyb5784 committed:
Merge branch 'in-toto:develop' into feature/cicd-integration-planner
